### PR TITLE
Fix httpworker indexing in webhost

### PR DESCRIPTION
**Pull request by ankitkumarr**
Fixes https://github.com/Azure/azure-functions-host/issues/6053

TODO: Need to cherry-pick for dev later

### CODE REVIEW

File: src/WebJobs.Script/Host/FunctionMetadataManager.cs [hunk 1]
```diff
@@ -42,7 +43,14 @@ public class FunctionMetadataManager : IFunctionMetadataManager
...
             // Every time script host is re-intializing, we also need to re-initialize
             // services that change with the scope of the script host.
-            scriptHostManager.HostInitializing += (s, e) => InitializeServices();
+            scriptHostManager.HostInitializing += (s, e) =>
+            {
+                InitializeServices();
+
+                // Once ScriptHost has initialized, functionMetadata has loaded the new services,
+                // it now works under the context of ScriptHost using the ScriptHost level config
+                _isWebHostMode = false;
```
Comment (pragnagopa) [inline src/WebJobs.Script/Host/FunctionMetadataManager.cs]: Nit: Rename to JobHostServicesInitialized?

**Review by pragnagopa — DISMISSED** [review]
Please do send a PR for V3 as well

**Comment by pragnagopa** [comment]
Pinging to get this merged before we generate CLI. Would be good to have this fix for the CLI release.

File: src/WebJobs.Script/Host/FunctionMetadataManager.cs [hunk 2]
```diff
@@ -23,6 +23,7 @@ public class FunctionMetadataManager : IFunctionMetadataManager
         private readonly IFunctionMetadataProvider _functionMetadataProvider;
         private bool _isHttpWorker;
         private bool _servicesReset = false;
+        private bool _jobHostServicesInitialized = true;
```
Comment (pragnagopa) [inline src/WebJobs.Script/Host/FunctionMetadataManager.cs]: this should start at false? and then set to true in HostInitializing the call back right?

**Review by pragnagopa — APPROVED** [review]

File: src/WebJobs.Script/Host/FunctionMetadataManager.cs (hunk 2 above)
Comment (ankitkumarr) [reply src/WebJobs.Script/Host/FunctionMetadataManager.cs]: Forgot to reverse booleans with name change haha. Fixed.

**Pull request merged**